Automattic/woocommerce-payments
# feat: maybe go to connect page when onboarding

Fixes #

#### Changes proposed in this Pull Request









It can happen that the account is not connected yet and we display the onboarding task.
Since the onboarding task is used in multiple places, I'm putting a redirect check there, in case the account is not connected.

#### Testing instructions





- Set this up by either:
  - Having a brand new installation
  - Setting the "Force the plugin to act as disconnected from WCPay" flag in the dev tools
- Go to WC Home http://localhost:8082/wp-admin/admin.php?page=wc-admin
- Either:
  - The "Boost your sales" task should not be displayed
  - Click on the "Boost your sales" inbox notice (only visible when you have UPE disabled and the account had been connected before)
![Screen Shot 2021-08-04 at 4 53 33 PM](https://user-images.githubusercontent.com/273592/128260156-e1bda3d7-0457-4114-af8c-723e12c89ff4.png)
    - You should be redirected to the connect page
- Once the account is connected, you should be allowed to go through onboarding

To recap:
- If account has not gone through account connection (i.e.: it is a new WCPay installation):
  - Onboarding task is not displayed
  - Inbox note is not displayed
- If account has been connected before, but it got disconnected (testable by setting the "Force the plugin to act as disconnected from WCPay" flag in the dev tools)
![Screen Shot 2021-08-05 at 9 28 49 AM](https://user-images.githubusercontent.com/273592/128374829-6d711ff6-683e-4a5e-8fdb-b554612afe06.png)
  - Onboarding task is not displayed
  - Inbox note _could_ be displayed if the merchant hasn't set up UPE yet (test by un-checking the UPE flag in Dev Tools)
    - If it is displayed and clicked, it redirects to account connection page with an error message
![Screen Shot 2021-08-05 at 9 29 12 AM](https://user-images.githubusercontent.com/273592/128374883-ecb2f1bb-e4af-4ca7-8fb4-95d12fb7f555.png)
- If account is connected
  - Onboarding task is not displayed
  - Inbox note is displayed
![Screen Shot 2021-08-05 at 9 26 44 AM](https://user-images.githubusercontent.com/273592/128374944-5083a0f3-a851-4691-8116-ec180180c4f1.png)

-------------------
